GameServer.java

```diff
@@ -99,180 +103,63 @@ private void processNickname(Spieler pClient, String pMessage){
+    private void processAuthorized(Spieler pClient, String pMessage) {
         String clientIP = pClient.gibIP();
         int clientPort = pClient.gibPort();
-        Spiel s = gibSpielNachSpieler(pClient);
-        Spieler gegenspieler = s.gibGegenspieler(pClient);
         if (pMessage.equals("QUIT")) {
             send(clientIP, clientPort, "+OK see you soon");
             try {
-                send(gegenspieler.gibIP(), gegenspieler.gibPort(), "+WON");
-                gegenspieler.setzeZustand(OVER);
-                registriereGewinnerInDB(gegenspieler);
+                //TODO Spiele beenden bzw. adjournen
+                pClient.setzeZustand(DISCONNECTED);
             } catch (Exception e) {
                 System.out.println("Gegenspieler nicht mehr vorhanden!");
             }
-            s.loescheSpieler(pClient);
-            if (s.beideSpielerWeg()) loescheSpielAusListe(s);
-            loescheSpieler(pClient);
+            if (pClient.isGuest()) loescheSpieler(pClient);
             closeConnection(clientIP, clientPort);
-        } else {
-            send(clientIP, clientPort, "ERR it is not your turn");
-        }
-    }
-
-    /**
-     * Nachricht des Spielers, der am Zug ist, wird verarbeitet.
-     * @param pClient anfragender Spieler
-     * @param pMessage Nachricht des Spielers
-     */
-    private void processActive(Spieler pClient, String pMessage) {
-        String clientIP = pClient.gibIP();
-        int clientPort = pClient.gibPort();
-        String[] stuecke = pMessage.split(" ");
-        char symbol = pClient.gibSymbol();
-        Spiel s = gibSpielNachSpieler(pClient);
-        Spieler gegenspieler = s.gibGegenspieler(pClient);
-        boolean validInt = true;
-        int vonZeile=0, vonSpalte=0;
-        int bisZeile=0, bisSpalte=0;
-        int nachZeile=0, nachSpalte=0;
-        Position von = null, bis = null;
-        Vektor richtung = null;
-        if (stuecke.length == 7) {
-            if (stuecke[0].equals("SHIFT")) {
-                try {
-                    vonZeile = Integer.parseInt(stuecke[1]);
-                    vonSpalte = Integer.parseInt(stuecke[2]);
-                    bisZeile = Integer.parseInt(stuecke[3]);
-                    bisSpalte = Integer.parseInt(stuecke[4]);
-                    nachZeile = Integer.parseInt(stuecke[5]);
-                    nachSpalte = Integer.parseInt(stuecke[5]);
-                    von = new Position(vonSpalte, vonZeile);
-                    bis = new Position(bisSpalte, bisZeile);
-                    richtung = new Vektor(bisSpalte, bisZeile, nachSpalte, nachZeile);
-                } catch (Exception e) {
-                    //System.out.println("Konnte keine Integer parsen!");
-                    send(clientIP, clientPort, "Die Eingaben waren ung�ltig");
-                    validInt = false;
-                }
-                if (s != null && validInt) {
-                    boolean status = s.schiebe(von, bis, richtung,pClient);

```

Comment:
Hallo Heiko,

ups - ich hatte noch nicht gesehen, dass du die Methode schon auf den
AbbaloneServer angepasst hattest...

Die sollten zumindest als Kommentar drinbleiben.

Bin mir auch nicht sicher, ob meine Weiterentwicklung nicht zuuuu
schnell ist.

Gruß
Peter

Am 18.07.2016 um 19:30 schrieb HeikoDudzus:
> In GameServer.java
> <https://github.com/HeikoDudzus/Abalone/pull/8#discussion_r71191313>:
> 
>> -                    vonZeile = Integer.parseInt(stuecke[1]);
>> -                    vonSpalte = Integer.parseInt(stuecke[2]);
>> -                    bisZeile = Integer.parseInt(stuecke[3]);
>> -                    bisSpalte = Integer.parseInt(stuecke[4]);
>> -                    nachZeile = Integer.parseInt(stuecke[5]);
>> -                    nachSpalte = Integer.parseInt(stuecke[5]);
>> -                    von = new Position(vonSpalte, vonZeile);
>> -                    bis = new Position(bisSpalte, bisZeile);
>> -                    richtung = new Vektor(bisSpalte, bisZeile, nachSpalte, nachZeile);
>> -                } catch (Exception e) {
>> -                    //System.out.println("Konnte keine Integer parsen!");
>> -                    send(clientIP, clientPort, "Die Eingaben waren ung�ltig");
>> -                    validInt = false;
>> -                }
>> -                if (s != null && validInt) {
>> -                    boolean status = s.schiebe(von, bis, richtung,pClient);
> 
> Mir ist nicht klar, wieso die Aufrufe an die Spiel-Logik hier ersatzlos
> gestrichen sind. Ich sehe diesbezüglich nur Deletions
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/HeikoDudzus/Abalone/pull/8/files/dc244c19c41c88ca25877a7b6ab5dc940610b864..e787f5b5eba8162751773aa608f01b638b7c8eeb#r71191313>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATf65jSXVcq4IHakbjQHJmARTyLvkFM2ks5qW7hQgaJpZM4JOEvP>.
> 

-- 

Peter Scholl
Rotdornweg 34
53721 Siegburg
Telefon: 02241/147747
e-mail : scholl@unterrichtsportal.org